### DISCUSSION TITLE
Great Project - QQ

### DISCUSSION THREAD

**Discussion by stevenAthompson** [sipper69/Control3#8] [Q&A] [answered]
Excellent idea, and excellent project. It's already very usable.

Quick question: Is the code in github the most current? The MS store version offers hotkey support and the version I just built from the GH repo does not.

I'd like to add an option to rotate the screen for portrait mode, which sounds simple enough, but I don't think I'm working with the current version here and I wouldn't want to have to write it twice.

**Comment by sipper69** [comment, marked as answer]
Hi Steven.
No, that version is closed for a possible other project. 

Interesting idea! Rotation can be very handy. 

But effort spending on catching a hotkey is marginal compared to adding this complete new functionality, right?

**Comment by stevenAthompson** [comment]
No worries @sipper69 , I'll just bolt the hotkeys back on myself and maybe fork the project. Thanks for the great project and for the assistance! It's really coming in handy already.

**Comment by sipper69** [comment]
Great. 
I'll keep an eye on you!

**Comment by ArsenQ** [comment]
Cool idea, probably the only compiled option for Windows 10 based on ch9329.

I have a ch9329 board, rather than the cable as per the picture below. It comes with 4 jumpers: Mode0, Mode1, Cfg0, Cfg1.

Without me putting on any jumpers keyboard and mouse functionality doesn't seem to work, although the usb to serial adapter is detected and I see activity led (on usb-serial adapter) flashing when I move the mouse or press keyboard buttons.

![image](https://github.com/user-attachments/assets/1cfa1896-fd2d-4046-974a-8cc1b01f2aec)

**Reply by sipper69** [reply]
Hi.

Did you use the latest release from the Microsoft App store? It will scan on every usb serial cable no matter the type.

If the App gives connected in green it is able to get the active status of the ch9329. If it stays red it will not work. 

Maybe you should only connect Rx/tx/Gnd not the 5V. If you do, it is powered from two sides. You did try to swap Rx/Tx  positions right? Host-remote must be Tx to Rx and vice versa.

The 4 jumpers are used to set the right mode. They should all be 1. Don't know the default of that board, but you could try to put them all on to swap the state. Must be repowered after doing that.

Also possible that your CH9329 is not at the default baudrate (must be 9600). Only to solve manually via a serial console accepting hex strings. Or maybe the native tool, if you read Chinese. 

Cheers

**Reply by ArsenQ** [reply]
Thanks for the comprehensive answer.

I've used your app from the Microsoft App store.

I've done a bit more testing to make my query better: 
1. On a modern enough Dell laptop it works in bios and windows. 
2. On a modern enough HP desktop (Prodesk 600 g5) works in Ubuntu but doesn't work in bios. This issue could be related to the way ch9329 creates virtual usb hub(s) /w HID type devices (keyboard, mouse, etc), may be fixed /w adjustment of jumpers.

**Reply by sipper69** [reply]
Sounds you're right. My bet is in placing the 4 jumpers :-)
